### DISCUSSION TITLE
more artifact information in local scope

### DISCUSSION THREAD

**Discussion by sbowman-mitre** [cpm-cmake/CPM.cmake#296] [Ideas]
Hi, I love CPM 💯  and have started using it pervasively in my cmake projects. It has made such an improvement to my build descriptions and I have so much added control over my dependency map. Thank you!

I do have a feature request: add extra descriptive variables to local scope for each CPMAddPackage() call. Currently, you provide:
* <dependency>_SOURCE_DIR
* <dependency>_BINARY_DIR
* <dependency>_ADDED
For my build processes, I'd also like:
* <dependency>_VERSION
* <dependency>_BRANCH
* <dependency>_URL
* <dependency>_GIT_TAG

Basically, a local scope repeat of the inputs when CPMAddPackage() was called. That would allow my build processes to report the versions of dependencies that were defined in the build. There is a possible work-around, but it is awkward and I'd like your ideas. Should I add this as a new issue? Or do you have a recommendation for how to do this that I should try?

**Comment by TheLartians** [comment]
Hey @sbowman-mitre! Thanks for the kind feedback and suggestion! 

I see how the info can be useful. In case you haven't seen it already, CPM.cmake currently stores the version for all dependencies that can be retrieved using `CPMGetPackageVersion(<dependency name> <output variable>)`. Besides that, we don't save any more information about packages in the cache, so there is currently no trivial way to get that information afaik. 

Not sure if it's helpful to you, but we do create some files in the build directory which you could use to restore the original args. Specifically, in `<build dir>/CPM_modules` each dependency will have a `find_<dependency name>.cmake` script that will invoke CPM.cmake with the expanded original args.

**Comment by sbowman-mitre** [comment]
`CPMGetPackageVersion(<dependency name> <output variable>)` -- I actually had not noticed this and it does satisfy my current needs. 

I'll look also at `find_<dependency name>.cmake` to see what I can do with that.

Thank you! 👍
